lib/Backend/IRBuilder.cpp

```diff
@@ -750,28 +751,41 @@ IRBuilder::Build()
             }
         }
 #endif
-        if (IsLoopBodyInTry() && m_lastInstr->GetDst() && m_lastInstr->GetDst()->IsRegOpnd() && m_lastInstr->GetDst()->GetStackSym()->HasByteCodeRegSlot())
+
+        if (IsLoopBodyInTry() && lastProcessedInstrForJITLoopBody != m_lastInstr)
         {
-            StackSym * dstSym = m_lastInstr->GetDst()->GetStackSym();
-            Js::RegSlot dstRegSlot = dstSym->GetByteCodeRegSlot();
-            if (!this->RegIsTemp(dstRegSlot) && !this->RegIsConstant(dstRegSlot))
+            FOREACH_INSTR_BACKWARD_EDITING_IN_RANGE(
```

Comment:
There is `m_stSlots` to prevent reset of the same `symId`, traverse in backward so only the later store of given `symId` takes effect. Or we could get stale result if the earlier store takes effect if traverse in forward order.

---
In reply to: [100936203](https://github.com/Microsoft/ChakraCore/pull/2518#discussion_r100936203) [](ancestors = 100936203)